planner/optimize.go

```diff
@@ -26,10 +27,10 @@ import (
 
 // Optimize does optimization and creates a Plan.
 // The node must be prepared first.
-func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, error) {
-	fp := plannercore.TryFastPlan(sctx, node)
+func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, []*expression.NamingForMySQLProtocol, error) {
+	fp, names := plannercore.TryFastPlan(sctx, node)
```

Comment:
Can we save the names in the fast plan rather than returning it? It seems a little strange that `Optimize` function returns a name list.

Reply:
Previously, I don't want to do this since this field only used when building plan. Not used in the later optimization. So i don't want to keep it in any of the plans.
But today i found that storing it in the plan operator can save many codes.
So i'll change it in next commit.